monai/transforms/regularization/dictionary.py

```diff
@@ -31,18 +38,24 @@ class MixUpd(MapTransform):
     def __init__(
         self, keys: KeysCollection, batch_size: int, alpha: float = 1.0, allow_missing_keys: bool = False
     ) -> None:
-        super().__init__(keys, allow_missing_keys)
+        MapTransform.__init__(self, keys, allow_missing_keys)
         self.mixup = MixUp(batch_size, alpha)
 
+    def set_random_state(self, seed: int | None = None, state: np.random.RandomState | None = None) -> MixUpd:
+        super().set_random_state(seed, state)
+        self.mixup.set_random_state(seed, state)
+        return self
+
     def __call__(self, data):
-        self.mixup.randomize()
-        result = dict(data)
-        for k in self.keys:
-            result[k] = self.mixup.apply(data[k])
-        return result
+        d = dict(data)
+        # all the keys share the same random state
+        self.mixup.randomize(None)
```

Comment:
Should we add a `randomize` keyword argument for this and the other `__call__` methods so we can only optionally call `randomize`?

Reply:
Do you mean always make these transforms generate the same results? If we want to do that, the better way may be introduce RandMixUp, RandCutMix, RandCutOut and make existing transforms not inherit from the `RandomizableTransform`. But it seems that these transforms were originally designed for augmentation, so maybe deterministic is not commonly used?

Reply:
OK, let's not do that then and see if anyone requests this change later.